In [2]:
import numpy as np
import pandas as pd

In [3]:
data = np.load('monthdata.npz')
totals = data['totals']
counts = data['counts']
print(data.files)
print(totals)
print(counts)


['totals', 'counts']
[[1809 1482 2159  374  606  428  442   68  956 1429 1349 1846]
 [ 121   22   46   40  683  616 2061  834  253  276   27  224]
 [ 896 1232  992 1106 1236 1015 1264 1096 1026 2422 1184 1099]
 [1491 1473 1452 1595 1001  725  734  445  845 2185 1305 1835]
 [ 930  561  533  204  546  748  662  274  785 1620 1293  295]
 [ 183  982  621  471  667  671  525 1599  433 1176  594  627]
 [ 363  491  729  681  346  269  399  667  663  452  497  821]
 [1682 1304 1616  242  516  582  328  138  784 2034 2402 1520]
 [ 220   90   28  120   98  306  154  280  488  158  176  116]]
[[30 22 30 30 21 21 20 25 20 19 17 16]
 [31 29 31 30 31 30 31 30 30 30 30 30]
 [31 29 31 30 30 30 31 31 30 31 30 31]
 [31 29 31 30 31 30 31 31 29 30 30 31]
 [31 29 31 29 31 29 30 31 30 31 30 29]
 [31 29 31 30 31 30 28 31 30 30 29 31]
 [31 28 31 30 31 30 31 31 30 31 29 31]
 [30 28 31 30 30 29 30 31 30 31 30 30]
 [31 28 31 30 31 30 31 31 30 30 30 30]]


In [21]:
np.sum(totals, axis = 1)

array([12948,  5203, 14568, 15086,  8451,  8549,  6378, 13148,  2234],
      dtype=int64)

# the 9th city has the lowest total precipitation over the year at 2234

In [4]:
np.argmin(np.sum(totals, axis = 1))

8

# index 8 or the 9th row

In [13]:
np.sum(totals, axis = 0) / np.sum(counts, axis = 0)

array([27.77978339, 30.42629482, 29.41007194, 17.96654275, 21.34456929,
       20.69498069, 24.97718631, 19.85661765, 24.06563707, 44.68441065,
       34.61568627, 32.36679537])

In [44]:
temp = np.sum(totals, axis = 1) / np.sum(counts, axis = 1)
print(temp)


[47.77859779 14.33333333 39.91232877 41.44505495 23.4099723  23.68144044
 17.52197802 36.52222222  6.15426997]


In [29]:
quaterly = np.reshape(totals, (36, 3))

In [32]:
sumQuaterly = np.sum(quaterly, axis = 1)

In [33]:
np.reshape(sumQuaterly, (9,4))

array([[5450, 1408, 1466, 4624],
       [ 189, 1339, 3148,  527],
       [3120, 3357, 3386, 4705],
       [4416, 3321, 2024, 5325],
       [2024, 1498, 1721, 3208],
       [1786, 1809, 2557, 2397],
       [1583, 1296, 1729, 1770],
       [4602, 1340, 1250, 5956],
       [ 338,  524,  922,  450]], dtype=int64)

In [5]:
totals = pd.read_csv('totals.csv').set_index(keys=['name'])
print(totals)

                                2016-01  2016-02  2016-03  2016-04  2016-05  \
name                                                                          
BURNABY SIMON FRASER U             1809     1482     2159      374      606   
CALGARY INTL A                      121       22       46       40      683   
GANDER INTL A                       896     1232      992     1106     1236   
HALIFAX INTL A                     1491     1473     1452     1595     1001   
REVELSTOKE                          930      561      533      204      546   
SHERBROOKE                          183      982      621      471      667   
TORONTO LESTER B. PEARSON INT'      363      491      729      681      346   
VANCOUVER INTL A                   1682     1304     1616      242      516   
YELLOWKNIFE A                       220       90       28      120       98   

                                2016-06  2016-07  2016-08  2016-09  2016-10  \
name                                               